src/lib.rs

```diff
@@ -260,6 +275,75 @@ impl Pixels {
     pub fn get_frame(&mut self) -> &mut [u8] {
         &mut self.pixels
     }
+
+    /// Calculate the pixel location from a physical location on the window,
+    /// dealing with window resizing, scaling, and margins. Takes a physical
+    /// position (x, y) within the window, and returns a pixel position (x, y).
+    ///
+    /// The location must be given in physical units (for example, winit's `PhysicalLocation`)
+    ///
+    /// If the given physical position is outside of the drawing area, this
+    /// function returns an `Err` value with the pixel coordinates outside of
+    /// the screen, using isize instead of usize.
+    ///
+    /// ```no_run
+    /// # use pixels::Pixels;
+    /// # let surface = wgpu::Surface::create(&pixels_mocks::RWH);
+    /// # let surface_texture = pixels::SurfaceTexture::new(1024, 768, surface);
+    /// const WIDTH:  u32 = 320;
+    /// const HEIGHT: u32 = 240;
+    ///
+    /// let mut pixels = Pixels::new(WIDTH, HEIGHT, surface_texture)?;
+    ///
+    /// // A cursor position in physical units
+    /// let cursor_position: (f64, f64) = winit::dpi::PhysicalPosition::new(0, 0).into();
+    ///
+    /// // Convert it to a pixel location
+    /// let pixel_position: (usize, usize) = pixels.window_pos_to_pixel(cursor_position)
+    ///     .unwrap_or_else(|(x, y)| ( // Clamp the output to within the screen
+    ///         x.max(0).min(WIDTH as isize - 1) as usize,
+    ///         y.max(0).min(HEIGHT as isize - 1) as usize,
+    ///     ));
+    /// # Ok::<(), pixels::Error>(())
+    /// ```
+    pub fn window_pos_to_pixel(
+        &self,
+        physical_position: (f64, f64),
+    ) -> Result<(usize, usize), (isize, isize)> {
+        let physical_size = (
+            self.surface_texture.width as f32,
+            self.surface_texture.height as f32,
+        );
+
+        let pixels_width = self.texture_extent.width as f32;
+        let pixels_height = self.texture_extent.height as f32;
+
+        let pos = ultraviolet::Vec4::new(
+            (physical_position.0 as f32 / physical_size.0 as f32 - 0.5) * pixels_width,
+            (physical_position.1 as f32 / physical_size.1 as f32 - 0.5) * pixels_height,
+            0.0,
+            1.0,
+        );
+
+        let new_pos = self.scaling_matrix_inverse * pos;
+
+        let new_pos = (
+            new_pos.x / new_pos.w + pixels_width / 2.0,
+            -new_pos.y / new_pos.w + pixels_height / 2.0,
+        );
+        let pixel_x = new_pos.0.floor() as isize;
+        let pixel_y = new_pos.1.floor() as isize;
```

Comment:
I think shadowing is a bit better for readability.

```suggestion
        let pos = self.scaling_matrix_inverse * pos;

        let pos = (
            pos.x / pos.w + pixels_width / 2.0,
            -pos.y / pos.w + pixels_height / 2.0,
        );
        let pixel_x = pos.0.floor() as isize;
        let pixel_y = pos.1.floor() as isize;
```